File: migrate_content.mjs

import fs from 'fs';
import path from 'path';

const PROJECTS_DIR = 'd:/Pet_Projects/astro-portfolio/src/content/projects';

const CATEGORIES = {
    "Vue": [
        "adamhall", "wortmann", "fishersports",
        "armedangels", "eterna", "babyone"
    ],
    "React": [
        "healthcare", "expensetracker", "nextfood", "tanevents", "placepicker",
        "feedback", "countdown", "food", "context", "restore"
    ],
    "Python": [
        "jobdata", "healthcare", "jobscraper", "jobdash", "moviedata"
    ]
};

function getTags(filename) {
    const basename = path.basename(filename, '.md');
    const tags = new Set();
    for (const [cat, projects] of Object.entries(CATEGORIES)) {
        if (projects.includes(basename)) {
            tags.add(cat);
        }
    }
    return Array.from(tags);
}

function processFile(filename) {
    const filepath = path.join(PROJECTS_DIR, filename);
    let content = fs.readFileSync(filepath, 'utf-8');

    // Extract Title
    // Format: # [Projects](/portfolio/) | Adam Hall ([Link]...)
    const titleMatch = content.match(/^#.*?\|\s*(.*?)\s*(\(\[Link\]|$)/m);
    let title = titleMatch ? titleMatch[1].trim() : path.basename(filename, '.md');
    
    // Clean up title if it contains extra markdown link syntax that wasn't caught
    title = title.replace(/\[|\]/g, '');

    // Extract Image
    const imgMatch = content.match(/(?:<img src="|!\[.*?\]\()(\.\.\/)?images\/(.*?)(?:"|\))/);
    let imagePath = "";
    if (imgMatch) {
        imagePath = `/images/${imgMatch[2]}`;
    }

    const tags = getTags(filename);

    // Remove the H1 title line
    const newContent = content.replace(/^#.*?\n/, '').trim();

    const frontmatter = `---
title: "${title}"
tags: ${JSON.stringify(tags)}
image: "${imagePath}"
weight: 0
---

`;

    fs.writeFileSync(filepath, frontmatter + newContent, 'utf-8');
    console.log(`Processed ${filename}: Title='${title}', Tags=[${tags}], Image='${imagePath}'`);
}

const files = fs.readdirSync(PROJECTS_DIR);
files.forEach(filename => {
    if (filename.endsWith('.md')) {
        processFile(filename);
    }
});

File: migrate_content.py

import os
import re

PROJECTS_DIR = r"d:\Pet_Projects\astro-portfolio\src\content\projects"

CATEGORIES = {
    "Vue": [
        "adamhall", "wortmann", "fishersports",
        "armedangels", "eterna", "babyone"
    ],
    "React": [
        "healthcare", "expensetracker", "nextfood", "tanevents", "placepicker",
        "feedback", "countdown", "food", "context", "restore"
    ],
    "Python": [
        "jobdata", "healthcare", "jobscraper", "jobdash", "moviedata"
    ]
}

def get_tags(filename):
    basename = os.path.splitext(filename)[0]
    tags = []
    for cat, projects in CATEGORIES.items():
        if basename in projects:
            tags.append(cat)
    return tags

def process_file(filename):
    filepath = os.path.join(PROJECTS_DIR, filename)
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()

    # Extract Title
    # Format: # [Projects](/portfolio/) | Adam Hall ([Link]...)
    title_match = re.search(r'^#.*?\|\s*(.*?)\s*(\(\[Link\]|$)', content, re.MULTILINE)
    title = title_match.group(1).strip() if title_match else filename.replace(".md", "").title()

    # Extract Image
    # Look for <img src="../images/..." /> or classic markdown ![...](...)
    # We need to adjust path from ../images/ to /images/
    img_match = re.search(r'(?:<img src="|!\[.*?\]\()(\.\./)?images/(.*?)(?:"|\))', content)
    image_path = ""
    if img_match: 
        # img_match.group(2) is the path after images/
        image_path = f"/images/{img_match.group(2)}"
    
    tags = get_tags(filename)
    
    # Remove the H1 title line as it's now in frontmatter
    new_content = re.sub(r'^#.*?\n', '', content, count=1).strip()

    frontmatter = f"""---
title: "{title}"
tags: {tags}
image: "{image_path}"
weight: 0
---

"""
    
    final_content = frontmatter + new_content
    
    with open(filepath, "w", encoding="utf-8") as f:
        f.write(final_content)
    
    print(f"Processed {filename}: Title='{title}', Tags={tags}, Image='{image_path}'")

for filename in os.listdir(PROJECTS_DIR):
    if filename.endswith(".md"):
        process_file(filename)
